modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/utils/ExternalLDAPServer.java

```diff
@@ -0,0 +1,632 @@
+import org.apache.directory.api.ldap.schema.manager.impl.DefaultSchemaManager;
+import org.apache.directory.server.constants.ServerDNConstants;
+import org.apache.directory.server.core.DefaultDirectoryService;
+import org.apache.directory.server.core.api.CoreSession;
+import org.apache.directory.server.core.api.DirectoryService;
+import org.apache.directory.server.core.api.InstanceLayout;
+import org.apache.directory.server.core.api.LdapPrincipal;
+import org.apache.directory.server.core.api.schema.SchemaPartition;
+import org.apache.directory.server.core.partition.impl.btree.jdbm.JdbmIndex;
+import org.apache.directory.server.core.partition.impl.btree.jdbm.JdbmPartition;
+import org.apache.directory.server.core.partition.ldif.LdifPartition;
+import org.apache.directory.server.ldap.LdapServer;
+import org.apache.directory.server.protocol.shared.transport.TcpTransport;
+import org.apache.directory.server.xdbm.Index;
+import org.apache.directory.shared.kerberos.KerberosAttribute;
+
+import java.io.BufferedInputStream;
+import java.io.BufferedOutputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+
+public class ExternalLDAPServer {
+
+    private DirectoryService directoryService;
+    private LdapServer server;
+    private String workingDirectoryPath;
+    private String schemaZipFilePath;
+    private int serverPort;
+
+    public static final String ADMIN_PASSWORD_ALGORITHM = "SHA";
+
+    public ExternalLDAPServer(String workingDirectory, String schemaZipFile, int serverPort) {
+
+        this.workingDirectoryPath = workingDirectory;
+        this.schemaZipFilePath = schemaZipFile;
+        this.serverPort = serverPort;
+    }
+
+    public void init() throws Exception {
+
+        initiateDirectoryService();
+        initiateLDAPServer();
+    }
+
+    public void startServer(boolean addDefaultPartition) throws Exception {
+
+        server.start();
+        changeConnectionUserPassword("admin");
+
+        if (addDefaultPartition) {
+            addPartition(workingDirectoryPath);
+        }
+    }
+
+    public void stopServer() throws Exception {
+
+        server.stop();
+    }
+
+    private void initiateDirectoryService() throws Exception {
+
+        directoryService = new DefaultDirectoryService();
+
+        InstanceLayout instanceLayout = new InstanceLayout(workingDirectoryPath);
+        directoryService.setInstanceLayout(instanceLayout);
+
+        initiateSchema();
+
+        directoryService.getChangeLog().setEnabled(false);
+        directoryService.setDenormalizeOpAttrsEnabled(true);
+
+        initSystemPartition();
+
+        directoryService.startup();
```

Comment:
Shall we remove unnecessary new lines?

Reply:
Removed unnecessary new lines here and in similar places.